lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -731,8 +731,12 @@ DefConditionalI('\ifinner', undef, sub { 0; });
 
 DefConditionalI('\ifmmode', undef, sub { LookupValue('IN_MATH'); });
 
-DefConditional('\if XToken XToken',    sub { $_[1]->getCharcode == $_[2]->getCharcode; });
-DefConditional('\ifcat XToken XToken', sub { $_[1]->getCatcode == $_[2]->getCatcode; });
+DefConditional('\if XToken XToken', sub {
+    ($_[1]->is_dont_expand == $_[2]->is_dont_expand) &&
+      ($_[1]->getCharcode == $_[2]->getCharcode); });
+DefConditional('\ifcat XToken XToken', sub {
+    ($_[1]->is_dont_expand == $_[2]->is_dont_expand) &&
+      ($_[1]->getCatcode == $_[2]->getCatcode); });
```

Comment:
Something like this seems necessary, but your testcase below doesn't seem to test it (at least, it passes w/o this change). I think the interesting thing about these ifs is their treatment of active chars.

Reply:
Indeed, I need to add more tests, I'm also feeling a bit paranoid about `\if` and `\ifcat` interacting with the dont_expand flags.